test/e2e_node/gci-init.sh

```diff
@@ -24,25 +24,23 @@ if [ -f "${CONTAINERD_HOME}/${CONTAINERD_ENV_METADATA}" ]; then
   source "${CONTAINERD_HOME}/${CONTAINERD_ENV_METADATA}"
 fi
 
-if [ "${CONTAINERD_CGROUPV2:-"false"}"  == "true" ]; then
-  # check cos image
-  if [ -r /etc/os-release ]; then
-    OS_ID="$(. /etc/os-release && echo "$ID")"
-  fi
-  if [ "${OS_ID}" = "cos" ]; then
-    if ! grep -q 'systemd.unified_cgroup_hierarchy=true' /proc/cmdline && [ "$(stat -fc %T /sys/fs/cgroup/)" != "cgroup2fs" ]; then
-      echo "Setting up cgroupv2"
-
-      mount_path="/tmp/esp"
-      mkdir -p "${mount_path}"
-      esp_partition="/dev/sda12"
-      mount "${esp_partition}" "${mount_path}"
-      sed -i 's/systemd.unified_cgroup_hierarchy=false/systemd.unified_cgroup_hierarchy=true/g' "${mount_path}/efi/boot/grub.cfg"
-      umount "${mount_path}"
-      rmdir "${mount_path}"
-
-      echo "Reconfigured grub; rebooting..."
-      reboot
+if [[ -n "${CONTAINERD_COS_CGROUP_MODE}" ]]; then
+  if uname -a | grep -q cos; then
```

Comment:
Is there a particular reason for switching away from the `ID` line parsing from `/etc/os-release` that was used before?  That's a much more standardized way to detect the specific Linux distribution than checking `uname -a` (which contains user-controlled fields such as the hostname, and could include the string "cos" even if it isn't COS).

Reply:
good point, there's no reason to switch to `uname -a` (I copied this over from another script used for testing). Switched back to `/etc/os-release` check as you suggested.